code/game/machinery/computer/medical.dm

```diff
@@ -258,20 +258,16 @@
 				src.active2 = null
 
 			else if(href_list["vir"])
-				var/type = href_list["vir"]
-				var/datum/disease/Dis = new type(0)
-				var/AfS = ""
-				for(var/mob/M in Dis.viable_mobtypes)
-					AfS += " [initial(M.name)];"
-				src.temp = {"<b>Name:</b> [Dis.name]
-<BR><b>Number of stages:</b> [Dis.max_stages]
-<BR><b>Spread:</b> [Dis.spread_text] Transmission
-<BR><b>Possible Cure:</b> [(Dis.cure_text||"none")]
-<BR><b>Affected Lifeforms:</b>[AfS]
+				var/href_type = text2path(href_list["vir"])
+				var/datum/disease/type = href_type
+				src.temp = {"<b>Name:</b> [initial(type.name)]
```

Comment:
```suggestion
				var/href_type = text2path(href_list["vir"])
				if(ispath(href_type, /datum/disease))
					  var/datum/disease/type = href_type
					  src.temp = {"<b>Name:</b> [initial(type.name)]
```
I think this would runtime if the href is not a disease path, might be good to prevent that while you're touching this.